cedar-policy-core/src/ast/value.rs

```diff
@@ -26,58 +27,255 @@ use thiserror::Error;
 
 /// This describes all the values which could be the dynamic result of evaluating an `Expr`.
 /// Cloning is O(1).
-#[derive(Debug, Clone, PartialOrd, Ord, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(into = "Expr")]
 #[serde(try_from = "Expr")]
 pub enum Value {
```

Comment:
Does it make sense to have a struct `Value`, which has a `ValueKind` and a `Loc`?

Reply:
We could do it that way.  I felt that in this case, the benefit of being able to `match value { Value::Lit { lit, .. } }` is better ergonomics for ignoring the `Loc` than something like `match value.value_kind() { ValueKind::Lit(lit) }` would be, but I'm open to the `ValueKind` approach too.

A more involved change, which I'd also be wiling to consider, would be renaming `Node<T>` to `Located<T>` or something, since it's really just a general wrapper around a `T` that adds a `Loc`.  Then we could leave `Value` alone but use `Located<Value>` in places where we expect a `Value` with optional `Loc` attached.  

Thoughts on either of these?

Reply:
For our _large_ enums I definitely prefer the `*Kind` pattern. The particular benefits I see are that it avoids redundancy and facilitates attaching extra meta-data later, but I agree that attaching it inline is more ergonomic when there number of variants is small. The `Value` enum is small enough that I'd have probably used the inline approach as well.

There is some value in standardizing on one pattern, but IMO it's not critical, and it may be better to just use whichever is nicer in a given situation. 

The `Located` option is an improvement over re-implementing the `*Kind` pattern everywhere, but it might not work out so cleanly given that `Node` and `ToAstError` currently have non-optional `Loc`s, and the `Expr` struct also stores the `data` entry used to type annotation.

Reply:
Discussed offline with the Cedar team and we decided to prefer the `*Kind` pattern in general.  I will adapt this PR after the holidays.

Reply:
Completed this change.